File: SdkTestAutomation.CliWrappers/SdkTestAutomation.CSharp/Operations/WorkflowOperations.cs
using System.Text.Json;
using Conductor.Api;
using Conductor.Client;
using SdkTestAutomation.Sdk.Models;
using SdkTestAutomation.CSharp.Extensions;

namespace SdkTestAutomation.CSharp.Operations;

public static class WorkflowOperations
{
    public static SdkResponse Execute(string operation, Dictionary<string, JsonElement> parameters)
    {
        return OperationUtils.ExecuteWithErrorHandling(() =>
        {
            var config = OperationUtils.CreateSdkConfiguration();
            var workflowApi = new WorkflowResourceApi(config);
            
            return operation switch
            {
                "get-workflow" => GetWorkflow(parameters, workflowApi),
                _ => throw new ArgumentException($"Unknown workflow operation: {operation}")
            };
        });
    }
    
    private static SdkResponse GetWorkflow(Dictionary<string, JsonElement> parameters, WorkflowResourceApi workflowApi)
    {
        var workflowId = parameters.GetString("workflowId");
        var workflow = workflowApi.GetExecutionStatus(workflowId);
        return SdkResponse.CreateSuccess(workflow);
    }
} 

File: SdkTestAutomation.CliWrappers/SdkTestAutomation.Java/src/main/java/com/conductor/sdkwrapper/operations/WorkflowOperations.java
package com.conductor.sdkwrapper.operations;

import com.netflix.conductor.client.http.ConductorClient;
import com.netflix.conductor.client.http.WorkflowClient;
import com.netflix.conductor.common.run.Workflow;
import com.conductor.sdkwrapper.OperationUtils;
import com.conductor.sdkwrapper.SdkResponse;

import java.util.Map;

public class WorkflowOperations {
    
    public static SdkResponse execute(String operation, Map<String, Object> parameters) {
        return OperationUtils.executeWithErrorHandling(() -> {
            ConductorClient client = OperationUtils.createSdkConfiguration();
            WorkflowClient workflowApi = new WorkflowClient(client);
            
            return switch (operation) {
                case "get-workflow" -> getWorkflow(parameters, workflowApi);
                default -> throw new IllegalArgumentException("Unknown workflow operation: " + operation);
            };
        });
    }
    
    private static SdkResponse getWorkflow(Map<String, Object> parameters, WorkflowClient workflowApi) throws Exception {
        String workflowId = (String) parameters.get("workflowId");
        Workflow workflow = workflowApi.getWorkflow(workflowId, false);
        return SdkResponse.createSuccess(workflow);
    }
} 